File: 0055-jump-game/0055-jump-game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        n , max_reach = len(nums) - 1, 0
        
        for i in range(len(nums)):
            if i > max_reach:
                return False
            
            max_reach = max(max_reach, i + nums[i])
            
            if max_reach >= n:
                return True
            
        return False

File: 0055-jump-game/0055-jump-game.java
class Solution {
    public boolean canJump(int[] nums) {
        int n = nums.length - 1 , max_reach = 0;
        
        for (int i = 0; i <= n; i++) {
            if (i > max_reach) {
                return false;
            }
            
            max_reach = Math.max(max_reach, i + nums[i]);
            
            if (max_reach >= n) {
                return true;
            }
        }
        return false;
    }
}